packages/types/src/services/consentUi/utils/zodCustomErrors.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2023 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { z } from 'zod';
+
+const ZOD_CUSTOM_ERRORS = [
+	'guardianEmailMissing',
+	'guardianHasParticipantEmail',
+	'guardianHasParticipantPhoneNumber',
+	'guardianInfoMissing',
+	'participantEmailMissing',
+	'participantHasGuardianEmail',
+	'participantHasGuardianInfo',
+	'participantLessThanMinimumAge',
+	'participantMissingPhoneNumber',
+] as const;
+
+export const ZodCustomError = z.enum(ZOD_CUSTOM_ERRORS);
+export type ZodCustomError = z.infer<typeof ZodCustomError>;
```

Comment:
How would you feel about trying Zod's `z.nativeEnum` here: https://zod.dev/?id=native-enums

This would allow us to make the `ZOD_CUSTOM_ERRORS` an object with keys, so instead of using `ZodCustomError.enum.participantMissingPhoneNumber`, we can use `ZOD_CUSTOM_ERRORS.PARTICIPANT_MISSING_PHONE_NUMBER` when we want to reference a specific custom error.

```suggestion
export const ZOD_CUSTOM_ERRORS = [
	GUARDIAN_EMAIL_MISSING: 'guardianEmailMissing',
	GUARDIAN_HAS_PARICIPANT_EMAIL: 'guardianHasParticipantEmail',
	GUARDIAN_HAS_PARTICIPANT_PHONE_NUMBER: 'guardianHasParticipantPhoneNumber',
	GUARDIAN_INFO_MISSING: 'guardianInfoMissing',
	PARTICIPANT_EMAIL_MISSING: 'participantEmailMissing',
	PARTICIPANT_HAS_GUARDIAN_EMAIL: 'participantHasGuardianEmail',
	PARTICIPANT_HAS_GUARDIAN_INFO: 'participantHasGuardianInfo',
	PARTICIPANT_LESS_THAN_MINIMUM_AGE: 'participantLessThanMinimumAge',
	PARTICIPANT_MISSING_PHONE_NUMBER: 'participantMissingPhoneNumber',
} as const;

export const ZodCustomError = z.nativeEnum(ZOD_CUSTOM_ERRORS);
export type ZodCustomError = z.infer<typeof ZodCustomError>;
```

Reply:
After discussing with @anncatton and as per the Zod docs recommendation, you can ignore this suggestion and continue with the regular Zod `enum` instead of `nativeEnum`!